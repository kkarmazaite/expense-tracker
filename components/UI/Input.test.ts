import { mount } from "@vue/test-utils";
import { expect, describe, it } from 'vitest'
import Input from "./Input.vue";

describe('Input.vue', async() => {
  it('Should render', () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: 'value',
        placeholder: 'placeholder',
        type: 'text',
        label: 'label',
      },
    });
    expect(wrapper.find('label').text()).toBe('label')
    expect(wrapper.find('input').element.value).toBe('value')
    expect(wrapper.find('input').element.placeholder).toBe('placeholder')
    expect(wrapper.find('input').element.type).toBe('text')
    expect(wrapper.find('div:first-child')).toMatchSnapshot()
  })
})
