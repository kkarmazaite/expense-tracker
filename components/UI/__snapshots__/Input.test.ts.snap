// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input.vue > Should render 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <div
    class="w-full"
  >
    <label
      class="block pl-3 ml-px text-sm font-medium text-gray-700"
      for=""
    >
      label
    </label>
    <div
      class="mt-1"
    >
      <input
        class="block w-full px-4 border-b border-gray-300 focus:border-blue-500 sm:text-sm"
        placeholder="placeholder"
        type="text"
      />
    </div>
  </div>,
}
`;
